samples/extensions/raytracing_extended/raytracing_extended.cpp

```diff
@@ -0,0 +1,1450 @@
+	create_display_pipeline();
+	create_shader_binding_tables();
+	create_descriptor_sets();
+	build_command_buffers();
+
+	prepared = true;
+	return true;
+}
+
+void RaytracingExtended::draw()
+{
+	device->get_fence_pool().wait();
+	device->get_fence_pool().reset();
+	ASSERT_LOG(raytracing_command_buffers.size() == draw_cmd_buffers.size(), "The number of raytracing command buffers must match the render queue size")
+	ApiVulkanSample::prepare_frame();
+	size_t i = current_buffer;
+
+	VkSubmitInfo submit       = vkb::initializers::submit_info();
+	submit.commandBufferCount = 1;
+	submit.pCommandBuffers    = &raytracing_command_buffers[i];
+
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit, device->request_fence()));
+	device->get_fence_pool().wait();
+
+	VkCommandBufferBeginInfo begin = vkb::initializers::command_buffer_begin_info();
+	VK_CHECK(vkBeginCommandBuffer(draw_cmd_buffers[i], &begin));
+
+	VkImageSubresourceRange subresource_range = {VK_IMAGE_ASPECT_COLOR_BIT, 0, 1, 0, 1};
+	/*
+	    Copy ray tracing output to swap chain image
+	*/
+	// Prepare current swap chain image as transfer destination
+	vkb::set_image_layout(
+	    draw_cmd_buffers[i],
+	    get_render_context().get_swapchain().get_images()[i],
+	    VK_IMAGE_LAYOUT_UNDEFINED,
+	    VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
+	    subresource_range);
+
+	// Prepare ray tracing output image as transfer source
+	vkb::set_image_layout(
+	    draw_cmd_buffers[i],
+	    storage_image.image,
+	    VK_IMAGE_LAYOUT_GENERAL,
+	    VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
+	    subresource_range);
+
+	VkImageCopy copy_region{};
+	copy_region.srcSubresource = {VK_IMAGE_ASPECT_COLOR_BIT, 0, 0, 1};
+	copy_region.srcOffset      = {0, 0, 0};
+	copy_region.dstSubresource = {VK_IMAGE_ASPECT_COLOR_BIT, 0, 0, 1};
+	copy_region.dstOffset      = {0, 0, 0};
+	copy_region.extent         = {width, height, 1};
+	vkCmdCopyImage(draw_cmd_buffers[i], storage_image.image, VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
+	               get_render_context().get_swapchain().get_images()[i], VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL, 1, &copy_region);
+
+	// Transition swap chain image back for presentation
+	vkb::set_image_layout(
+	    draw_cmd_buffers[i],
+	    get_render_context().get_swapchain().get_images()[i],
+	    VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL,
+	    VK_IMAGE_LAYOUT_PRESENT_SRC_KHR,
+	    subresource_range);
+
+	// Transition ray tracing output image back to general layout
+	vkb::set_image_layout(
+	    draw_cmd_buffers[i],
+	    storage_image.image,
+	    VK_IMAGE_LAYOUT_TRANSFER_SRC_OPTIMAL,
+	    VK_IMAGE_LAYOUT_GENERAL,
+	    subresource_range);
+	VK_CHECK(vkEndCommandBuffer(draw_cmd_buffers[i]));
+
+	submit_info.commandBufferCount = 1;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit_info, device->request_fence()));
+	device->get_fence_pool().wait();
+	ApiVulkanSample::submit_frame();
+}
+
+void RaytracingExtended::draw_gui()
+{
+	gui->show_options_window(
```

Comment:
Currently when running the sample I cannot see the GUI, is that expected?

Reply:
until GUI's work in the framework, we took out the GUI here to resolve this.  However, will add it back when #334 is merged in.  Can do a separate PR for that addition.

Reply:
GUI is removed until #334 is merged.